src/NzbDrone.Core/Plugins/PluginVersion.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+
+namespace NzbDrone.Core.Plugins
+{
+    public class PluginVersion : IComparable<PluginVersion>, IEquatable<PluginVersion>
+    {
+        public static readonly string[] DefaultTree = new[] { "master", "main" };
+
+        public Version BaseVersion { get; }
+        public string Suffix { get; }
+        public bool HasSuffix => !string.IsNullOrWhiteSpace(Suffix);
+
+        public PluginVersion(Version version, string suffix = null)
+        {
+            BaseVersion = version ?? throw new ArgumentNullException(nameof(version));
+            Suffix = CleanSuffix(suffix);
+        }
+
+        public static bool IsDefaultTree(string treeName) =>
+            Array.Exists(DefaultTree, tree =>
+                string.Equals(tree, treeName, StringComparison.OrdinalIgnoreCase));
+
+        public static PluginVersion Parse(string versionString)
+        {
+            if (string.IsNullOrWhiteSpace(versionString))
+            {
+                return null;
+            }
+
+            var parts = versionString.TrimStart('v').Split('-', 2);
+            return Version.TryParse(parts[0], out var version) ? new PluginVersion(version, parts.Length > 1 ? parts[1] : null) : null;
+        }
+
+        public override string ToString() =>
+            HasSuffix ? $"{BaseVersion}-{Suffix}" : BaseVersion.ToString();
+
+        public string ToFormattedString()
+        {
+            var build = BaseVersion.Build == -1 ? 0 : BaseVersion.Build;
+            var revision = BaseVersion.Revision == -1 ? 0 : BaseVersion.Revision;
+            var baseVersionString = $"{BaseVersion.Major}.{BaseVersion.Minor}.{build}.{revision}";
+
+            return HasSuffix && !IsDefaultTree(Suffix)
+                ? $"{baseVersionString} ({Suffix})"
+                : baseVersionString;
+        }
+
+        public int CompareTo(PluginVersion other)
+        {
+            if (other is null)
+            {
+                return 1;
+            }
+
+            var baseComp = BaseVersion.CompareTo(other.BaseVersion);
+            if (baseComp != 0)
+            {
+                return baseComp;
+            }
+
+            return (Suffix, other.Suffix) switch
+            {
+                (null, not null) => 1,
+                (not null, null) => -1,
+                _ => string.Compare(Suffix, other.Suffix, StringComparison.OrdinalIgnoreCase)
```

Comment:
Since we're doing semver, does this take into account that an empty (not just null) suffix is above one with suffix, eg stable vs dev/rc?